core/domain/blog_domain.py

```diff
@@ -0,0 +1,609 @@
+                raise utils.ValidationError(
+                    'Expected thumbnail filename to be a string, received: %s.'
+                    % thumbnail_filename)
+
+        if thumbnail_filename == '':
+            raise utils.ValidationError(
+                'Thumbnail filename should not be empty.')
+
+        utils.require_valid_thumbnail_filename(thumbnail_filename)
+
+    def validate(self, strict=False):
+        """Validates various properties of the blog post object.
+
+        Args:
+            strict: bool. Enable strict checks on the blog post when the blog
+                post is published or is going to be published.
+
+        Raises:
+            ValidationError. One or more attributes of blog post are invalid.
+        """
+        self.require_valid_blog_post_id(self.id)
+        self.require_valid_title(self.title, strict)
+        self.require_valid_tags(self.tags, strict)
+        self.require_valid_thumbnail_filename(
+            self.thumbnail_filename, strict=strict)
+
+        if not isinstance(self.content, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected contents to be a string, received: %s' % self.content)
+
+        if strict:
+            self.require_valid_url_fragment(self.url_fragment)
+            if len(self.content) == 0:
+                raise utils.ValidationError('Content can not be empty')
+
+    @classmethod
+    def require_valid_blog_post_id(cls, blog_post_id):
+        """Checks whether the blog post ID is a valid one.
+
+        Args:
+            blog_post_id: str. The blog post ID to validate.
+        """
+        if not isinstance(blog_post_id, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Blog Post ID should be a string, received: %s' % blog_post_id)
+
+        if len(blog_post_id) != constants.BASE_MODELS_ID_LENGTH:
+            raise utils.ValidationError('Invalid Blog Post ID.')
+
+    @classmethod
+    def require_valid_tags(cls, tags, strict):
+        """Validates tags for the blog post object.
+
+        Args:
+            tags: list(str). The list of tags assigned to a blog post.
+            strict: bool. Enable strict checks on the blog post when the blog
+                post is published or is going to be published.
+
+        Raises:
+            ValidationErrors.
+        """
+        if not isinstance(tags, list):
+            raise utils.ValidationError(
+                'Expected \'tags\' to be a list, received: %s' % tags)
+
+        for tag in tags:
+            if not isinstance(tag, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Expected each tag in \'tags\' to be a string, received: '
+                    '\'%s\'' % tag)
+
+            if len(tag) == 0:
+                raise utils.ValidationError(
+                    'Tag should not be empty, received: \'\'')
```

Comment:
```suggestion
                    'Tag should not be empty.'
```

This^ much should be enough, right?

Reply:
Done.